web-client/src/components/span/span-detail-trace.tsx

```diff
@@ -15,33 +18,105 @@ export function SpanDetailTrace(props: {
   return (
-    <tr class="even:bg-nearly-invisible cursor-pointer hover:bg-[#ffffff05] even:hover:bg-[#ffffff10]">
-      <td class="py-1 px-4">{props.span.name}</td>
-      <td class="py-1 px-4 relative w-[60%]">
-        <div class="relative w-[90%]">
-          <div class="bg-gray-800 w-full absolute rounded-sm h-2" />
-          <div
-            class="relative h-2"
-            style={computeWaterfallStyle(
-              props.span,
-              props.durations.start,
-              props.durations.end,
-              props.durations.shortest,
-              props.durations.longest
-            )}
-          >
-            {/* Slices is "time slices" as in multiple entry points to a given span */}
-            <For each={computeSlices(props.span.original)}>
-              {(slice) => (
-                <div
-                  class="absolute bg-teal-500 top-0 left-0 h-full"
-                  style={slice}
-                />
+    <Popover.Root>
+      <Popover.Trigger
+        class="even:bg-nearly-invisible cursor-pointer hover:bg-[#ffffff05] even:hover:bg-[#ffffff10]"
+        as="tr"
+      >
+        <td class="py-1 px-4">{props.span.name}</td>
+        <td class="py-1 px-4 relative w-[70%]">
+          <div class="relative">
+            <div class="bg-gray-800 w-full absolute rounded-sm h-2" />
+            <div
+              class="relative h-2"
+              style={computeWaterfallStyle(
+                props.span,
+                props.durations.start,
+                props.durations.end,
+                props.durations.shortest,
+                props.durations.longest
               )}
-            </For>
+            >
+              {/* Slices is "time slices" as in multiple entry points to a given span */}
+              <For each={computeSlices(props.span.original)}>
+                {(slice) => <SpanDetailSlice slice={slice} />}
+              </For>
+            </div>
           </div>
-        </div>
-      </td>
-    </tr>
+        </td>
+        <td class="py-1 px-4">{props.span.time.toFixed(2)}ms</td>
+      </Popover.Trigger>
+      <Popover.Portal>
+        <SpanDetailPopOverContent span={props.span} />
+      </Popover.Portal>
+    </Popover.Root>
+  );
+}
+
+function SpanDetailPopOverContent(props: { span: UiSpan }) {
+  const busy = (span: Span) =>
+    span.enters.reduce((acc, enter, i) => {
+      const exit = span.exits[i];
+
+      return acc + (exit.timestamp - enter.timestamp);
+    }, 0);
+
+  return (
+    <Popover.Content
+      class="z-50 bg-gray-700 rounded-sm drop-shadow-2xl px-2 py-1 focus:outline-none grid grid-cols-2"
+      style={{ "max-width": "min(calc(100vw - 16px), 380px)" }}
+    >
+      <Popover.Arrow />
+      <span>{props.span.name}</span>
+      <span />
+      <span>Busy</span>
+      <span>{(busy(props.span.original) / 1e6).toFixed(3)}ms</span>
+      <span>Idle</span>
+      <span>
+        {(props.span.time - busy(props.span.original) / 1e6).toFixed(3)}ms
+      </span>
```

Comment:
This is tabular data being shown as a bunch of spans.
It's non-semantic code that makes things harder to style, less reusable, and not accessible.

Even it still looks the way it does now, the html tags should probably be `<table>`

Reply:
I made this a table now.